app/src/main/java/edu/cascadia/mobas/campusguidebook/data/repository/AppRepository.java

```diff
@@ -47,290 +44,118 @@ public static AppRepository getInstance() {
+    public void delete(Sustainability... sustainabilities) {
+        mAppExecutors.diskIO().execute( () -> mAppDatabase.runInTransaction( () -> {
+            for (Sustainability sustainability : sustainabilities) {
+                mAppDatabase.SustainabilityDao().delete(sustainability);
+            }
+        }));
     }
 
-    // we are creating a async task method to insert new Event.
-    private static class InsertSustainabilityAsyncTask extends AsyncTask<Sustainability, Void, Void> {
-        private SustainabilityDao Sustainabilitydao;
-
-        private InsertSustainabilityAsyncTask(SustainabilityDao dao) {
-            this.Sustainabilitydao = Sustainabilitydao;
-        }
-
-        @Override
-        protected Void doInBackground(Sustainability... model) {
-            // below line is use to insert our model in dao.
-            Sustainabilitydao.insert(model[0]);
-            return null;
-        }
+    // Additional User CRUD operations
+    public void insert(User... users) {
+        mAppExecutors.diskIO().execute( () -> mAppDatabase.runInTransaction( () -> {
+            for (User user : users) {
+                mAppDatabase.UserDao().insert(user);
+            }
+        }));
     }
-    // we are creating a async task method to update our Event.
-    private static class UpdateSustainabilityAsyncTask extends AsyncTask<Sustainability, Void, Void> {
-        private SustainabilityDao Sustainabilitydao;
-
-        private UpdateSustainabilityAsyncTask(SustainabilityDao Sustainabilitydao) {
-            this.Sustainabilitydao = Sustainabilitydao;
-        }
 
-        @Override
-        protected Void doInBackground(Sustainability... models) {
-            // below line is use to update
-            // our model in dao.
-            Sustainabilitydao.update(models[0]);
-            return null;
-        }
+    public void update(User... users) {
+        mAppExecutors.diskIO().execute( () -> mAppDatabase.runInTransaction( () -> {
+            for (User user : users) {
+                mAppDatabase.UserDao().update(user);
+            }
+        }));
     }
 
-    // we are creating a async task method to delete Event.
-    private static class DeleteSustainabilityAsyncTask extends AsyncTask<Sustainability, Void, Void> {
-        private SustainabilityDao Sustainabilitydao;
-
-        private DeleteSustainabilityAsyncTask(SustainabilityDao Sustainabilitydao) {
-            this.Sustainabilitydao = Sustainabilitydao;
-        }
-
-        @Override
-        protected Void doInBackground(Sustainability... models) {
-            // below line is use to delete
-            // our Event model in dao.
-            Sustainabilitydao.delete(models[0]);
-            return null;
-        }
+    public void delete(User... users) {
+        mAppExecutors.diskIO().execute( () -> mAppDatabase.runInTransaction( () -> {
+            for (User user : users) {
+                mAppDatabase.UserDao().delete(user);
+            }
+        }));
```

Comment:
Excellent generalization of CRUD operations coded for more than one entity at a time.  Beautiful work